### PR TITLE
Handle empty Optimo paragraph blocks

### LINKED ISSUE #9313: Optimo - Empty Blocks Messing with Design
**Describe the bug**
We've seen several examples now where editorial (or Optimo) insert additional empty blocks into their content.  Rendering these empty blocks can subvert our design, eg, the misaligned timestamp here: 

Screen Shot 2021-08-11 at 05 28 14

This is caused by the first text section starting with a paragraph which has no content:
Screen Shot 2021-08-11 at 05 30 18

Which we render without checking:
Screen Shot 2021-08-11 at 05 28 58

**To Reproduce**
Example asset: https://www.test.bbc.com/arabic/articles/cl5m3453w36o

**Expected behaviour**
TBC - requires product input

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Checklist**
- [ ] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

### PR DESCRIPTION
**Pull request by rebeccamcginn**
Resolves #9313

**Overall change:**
We are currently rendering empty paragraph blocks, which upsets the design of the page. This PR filters out the empty paragraph blocks.

**Code changes:**

- Added a transformer that filters out all empty paragraph blocks.
- Moved reusable test functions to testHelpers.js and reusable fixture data one dir up.
- Added tests for the transformer.

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by amoore108 — APPROVED** [review]

File: src/app/routes/article/handleEmptyParagraphBlocks/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import path from 'ramda/src/path';
+import assocPath from 'ramda/src/assocPath';
+
+const getBlocks = path(['content', 'model', 'blocks']);
+const setBlocks = assocPath(['content', 'model', 'blocks']);
+const setTextBlock = assocPath(['model', 'blocks']);
+
+const isEmptyParagraph = block => {
+  return block.type === 'paragraph' && !block.model.text;
```
Comment (ryanmccombe) [inline src/app/routes/article/handleEmptyParagraphBlocks/index.js]: Is model always defined?
```suggestion
  return block.type === 'paragraph' && !block.model?.text;
```

File: src/app/routes/article/handleEmptyParagraphBlocks/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+const removeEmptyParagraphsFromTextBlock = textBlock => {
+  const cleanedTextData = textBlock.model.blocks.filter(
```
Comment (ryanmccombe) [inline src/app/routes/article/handleEmptyParagraphBlocks/index.js]: same here, what if `model` or `model.blocks` are undefined?

**Review by ryanmccombe — APPROVED** [review]
👍

File: src/app/routes/article/handleEmptyParagraphBlocks/index.js (hunk 1 above)
Comment (rebeccamcginn) [reply src/app/routes/article/handleEmptyParagraphBlocks/index.js]: I think model and model.blocks will always be defined: https://www.test.bbc.com/arabic/articles/cl5m3453w36o.json
Paragraph contains a fragment block, even when empty

**Pull request merged**